env0/resource_template_test.go

```diff
@@ -0,0 +1,321 @@
+		mock.EXPECT().TemplateCreate(client.TemplateCreatePayload{
+			Name:             template.Name,
+			Repository:       template.Repository,
+			Type:             defaultType,
+			TerraformVersion: defaultVersion,
+		}).Times(1).Return(template, nil)
+		mock.EXPECT().TemplateDelete(template.Id).Return(nil)
+	})
+}
+
+func TestUnitTemplateResourceInvalidType(t *testing.T) {
+	template := client.Template{
+		Name:       "template0",
+		Repository: "env0/repo",
+	}
+
+	testCase := resource.TestCase{
+		ProviderFactories: testUnitProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
+					"name":       template.Name,
+					"repository": template.Repository,
+					"type":       "gruntyform",
+				}),
+				ExpectError: regexp.MustCompile(`'type' can either be 'terraform' or 'terragrunt'`),
+			},
+		},
+	}
+
+	runUnitTest(t, testCase, func(mock *client.MockApiClientInterface) {})
+}
+
+func TestUnitTemplateResourceSshKeys(t *testing.T) {
+	initialSshKey1 := client.TemplateSshKey{
+		Id:   "ssh-key-id-1",
+		Name: "ssh-key-name-1",
+	}
+	initialSshKey2 := client.TemplateSshKey{
+		Id:   "ssh-key-id-2",
+		Name: "ssh-key-name-2",
+	}
+	updatedSshKey1 := client.TemplateSshKey{
+		Id:   "updated-ssh-key-id-1",
+		Name: "updated-ssh-key-name-1",
+	}
+	updatedSshKey2 := client.TemplateSshKey{
+		Id:   "updated-ssh-key-id-2",
+		Name: "updated-ssh-key-name-2",
+	}
+
+	template := client.Template{
+		Id:               "id0",
+		Name:             "template0",
+		Repository:       "env0/repo",
+		TerraformVersion: defaultVersion,
+		Type:             string(defaultType),
+		SshKeys:          []client.TemplateSshKey{initialSshKey1, initialSshKey2},
+	}
+
+	updatedTemplate := client.Template{
+		Id:               template.Id,
+		Name:             template.Name,
+		Repository:       template.Repository,
+		TerraformVersion: defaultVersion,
+		Type:             string(defaultType),
+		SshKeys:          []client.TemplateSshKey{updatedSshKey1, updatedSshKey2},
+	}
+
+	testCase := resource.TestCase{
+		ProviderFactories: testUnitProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: sshKeyTemplateResourceConfig(template.Name, template.Repository, initialSshKey1, initialSshKey2),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceFullName, "id", template.Id),
+					resource.TestCheckResourceAttr(resourceFullName, "name", template.Name),
+					resource.TestCheckResourceAttr(resourceFullName, "repository", template.Repository),
+					resource.TestCheckResourceAttr(resourceFullName, "ssh_keys.0.id", initialSshKey1.Id),
+					resource.TestCheckResourceAttr(resourceFullName, "ssh_keys.0.name", initialSshKey1.Name),
+					resource.TestCheckResourceAttr(resourceFullName, "ssh_keys.1.id", initialSshKey2.Id),
+					resource.TestCheckResourceAttr(resourceFullName, "ssh_keys.1.name", initialSshKey2.Name),
```

Comment:
Can we DRY those like you did below?

Reply:
24e49d1c985ea36e7ac058f89846f74d8500c59e